### PR TITLE
Feat: [CDS-77217] : RBAC OPA policy and policyset onboarding (backend changes) 

### PR DESCRIPTION
**Pull request by yagyanshbhatia**
## Describe your changes 
- Onboarded OPA resources in resource groups
- Added permissions in a few managed roles

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51753)

### CODE REVIEW

File: 953-events-api/src/main/java/io/harness/eventsframework/EventsFrameworkMetadataConstants.java [hunk 1]
```diff
@@ -52,6 +52,10 @@ public final class EventsFrameworkMetadataConstants {
   public static final String USER_SCOPE_RECONCILIATION = "userScopeReconciliation";
   public static final String CHAOS_EXPERIMENT = "chaosexperiment";
   public static final String CHAOS_HUB = "chaoshub";
+
+  // Note that OPA_GOVERNANCE if changed, redis EntityTypeOPA needs to be changed in policy-mgmt as well.
+  public static final String OPA_GOVERNANCE_POLICY = "OPAPolicy";
+  public static final String OPA_GOVERNANCE_POLICYSET = "OPAPolicySet";
```
Comment (meenakshi-harness) [inline 953-events-api/src/main/java/io/harness/eventsframework/EventsFrameworkMetadataConstants.java:58]: Should we rename to opaPolicy and opaPolicySet ?

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 2]
```diff
@@ -767,6 +767,7 @@ roles:
         - core_service_access
         - core_environment_view
         - core_environment_access
+        - core_governancePolicySets_evaluate
```
Comment (meenakshi-harness) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:770]: increase version of this file.

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/privileged-roles.yml [hunk 3]
```diff
@@ -13,6 +13,8 @@ roles:
       - core_delegate_view
       - core_delegateconfiguration_view
       - core_pipeline_view
+      - core_governancePolicy_view
+      - core_governancePolicySets_view
```
Comment (meenakshi-harness) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/privileged-roles.yml:17]: same

File: platform-service/config/config.yml [hunk 4]
```diff
@@ -281,4 +281,11 @@ segmentConfiguration:
   enabled: false
   url: "segment_url_place_holder"
   apiKey: "segment_api_key"
-  certValidationRequired: false
\ No newline at end of file
+  certValidationRequired: false
+
+policyManagerSecret: secret
+
+opaClientConfig:
```
Comment (meenakshi-harness) [inline platform-service/config/config.yml:288]: don't we need replace config for these?

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/privileged-roles.yml (hunk 3 above)
Comment (yagyanshbhatia) [reply access-control/service/src/main/resources/io/harness/accesscontrol/roles/privileged-roles.yml:17]: done

File: platform-service/config/config.yml (hunk 4 above)
Comment (yagyanshbhatia) [reply platform-service/config/config.yml:288]: yes, my bad, adding it

File: 953-events-api/src/main/java/io/harness/eventsframework/EventsFrameworkMetadataConstants.java (hunk 1 above)
Comment (yagyanshbhatia) [reply 953-events-api/src/main/java/io/harness/eventsframework/EventsFrameworkMetadataConstants.java]: Done

**Comment by ashishsanodia-harness** [comment]
*[`dockerization/platform-service/scripts/replace_configs.sh` line 436 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewNwet5AaIFyxTUeHP:-NewNwet5AaIFyxTUeHQ:b-j62xuu) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/dockerization/platform-service/scripts/replace_configs.sh#L436)):*
> ```Shell
> replace_key_value opaClientConfig.baseUrl "$OPA_SERVER_BASEURL"
> if [[ "" != "$OPA_CONNECT_TIMEOUT" ]]; then
>   replace_key_value opaClientConfig.connectTimeOutSeconds "$OPA_CONNECT_TIMEOUT"
> ```

Why empty check? 
replace_key_value already doing empty check before replacing value.

**Comment by ashishsanodia-harness** [comment]
*[`platform-service/build/container/scripts/replace_configs.sh` line 323 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewOA0O9fEBsDV9W3jO:-NewOA0O9fEBsDV9W3jP:b-j62xuu) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/build/container/scripts/replace_configs.sh#L323)):*
> ```Shell
> if [[ "" != "$OPA_READ_TIMEOUT" ]]; then
>   replace_key_value opaClientConfig.readTimeOutSeconds "$OPA_READ_TIMEOUT"
> fi
> ```

Why empty check? 
replace_key_value already doing empty check before replacing value.

**Comment by ashishsanodia-harness** [comment]
*[`platform-service/chart/templates/config.yaml` line 30 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewODWw4hY7fLoEmZ6r:-NewODWw4hY7fLoEmZ6s:b-u6tvs2) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/chart/templates/config.yaml#L30)):*
> ```YAML
>   { { - if .Values.global.opa.enabled } }
>   OPA_SERVER_BASEURL: 'http://policy-mgmt.{{ .Release.Namespace }}.svc.cluster.local:3000/'
>   OPA_SERVER_SECRET: "dOkdsVqdRPPRJG31XU0qY4MPqmBBMk0PTAGIKM6O7TGqhjyxScIdJe80mwh5Yb5zF3KxYBHw6B3Lfzlq"
> ```

how this secret will get override?

**Comment by ashishsanodia-harness** [comment]
*[`platform-service/service/src/main/java/io/harness/platform/PlatformConfiguration.java` line 122 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewOauN9oWzZnZAMXk6:-NewOauN9oWzZnZAMXk7:b6hv1xa) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/service/src/main/java/io/harness/platform/PlatformConfiguration.java#L122)):*
> ```Java
>   @ConfigSecret private String policyManagerSecret;
>   private ServiceHttpClientConfig opaClientConfig;
>   @JsonProperty("enforcementClientConfiguration") private EnforcementClientConfiguration enforcementClientConfiguration;
> ```

No @JsonProperty mapped for these configurations?

**Comment by ashishsanodia-harness** [comment]
*[`platform-service/service/src/main/java/io/harness/platform/PlatformConfiguration.java` line 121 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewOnmt5ujlpW468Nop:-NewOnmuAA2MC6mlJJwc:b6hv1xa) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/service/src/main/java/io/harness/platform/PlatformConfiguration.java#L121)):*
> ```Java
>   private AccessControlClientConfiguration accessControlClientConfig;
>   @ConfigSecret private String policyManagerSecret;
>   private ServiceHttpClientConfig opaClientConfig;
> ```

No @JsonProperty mapped for these configurations?

**Comment by ashishsanodia-harness** [comment]
*[`platform-service/service/src/main/resources/localhost.pem` line 21 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewOrjJ5debhbl7pcOF:-NewOrjJ5debhbl7pcOG:bdfrdvi) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/service/src/main/resources/localhost.pem#L21)):*
> ```
> A59Y218O3g/9MH3UVc2J7UB68N41GzYxwjk7EQn6P10EaVUqTwtjXfAxEI84UXpj
> SBXC7VCUjCXqgsHIoGyWhBY70P04
> -----END CERTIFICATE-----
> ```

Why this file?

File: platform-service/chart/templates/config.yaml [hunk 5]
```diff
@@ -25,6 +25,10 @@ data:
   RESOURCE_GROUP_CLIENT_BASE_URL: 'http://platform-service.{{ .Release.Namespace }}.svc.cluster.local:9005/api/'
   NG_MANAGER_CLIENT_BASEURL: 'http://ng-manager:7090/'
   PIPELINE_SERVICE_CLIENT_BASEURL: 'http://pipeline-service:12001/api/'
+  { { - if .Values.global.opa.enabled } }
```
Comment (arya-harness) [inline platform-service/chart/templates/config.yaml:28]: { { .. } } will not render
use {{ - if .Values.global.opa.enabled }} instead. 
Please confirm by templating the charts by running `helm template .` on the charts directory(wherever the Chart.yaml reside)

File: platform-service/chart/templates/config.yaml [hunk 6]
```diff
@@ -25,6 +25,10 @@ data:
...
+  OPA_SERVER_BASEURL: 'http://policy-mgmt.{{ .Release.Namespace }}.svc.cluster.local:3000/'
+  OPA_SERVER_SECRET: "dOkdsVqdRPPRJG31XU0qY4MPqmBBMk0PTAGIKM6O7TGqhjyxScIdJe80mwh5Yb5zF3KxYBHw6B3Lfzlq"
+  { { - end } }
```
Comment (arya-harness) [inline platform-service/chart/templates/config.yaml:31]: same here

File: platform-service/chart/values.yaml [hunk 7]
```diff
@@ -25,7 +27,7 @@ global:
   useExternalSecrets: false
   loadbalancerURL: "http://test/abc"
   imagePullSecrets: []
-  stackDriverLoggingEnabled: false
+  stackDriverLoggingEnabled: true
```
Comment (arya-harness) [inline platform-service/chart/values.yaml:30]: by default, please set it as false.

File: platform-service/service/src/main/resources/localhost.pem [hunk 8]
```diff
@@ -0,0 +1,21 @@
+-----BEGIN CERTIFICATE-----
```
Comment (arya-harness) [inline platform-service/service/src/main/resources/localhost.pem:1]: what is the requirement for this file? gitleaks might flag it later

**Comment by yagyanshbhatia** [comment]
> _[`dockerization/platform-service/scripts/replace_configs.sh` line 436 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewNwet5AaIFyxTUeHP:-NewNwet5AaIFyxTUeHQ:b-j62xuu) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/dockerization/platform-service/scripts/replace_configs.sh#L436)):_
> 
> > ```shell
> > replace_key_value opaClientConfig.baseUrl "$OPA_SERVER_BASEURL"
> > if [[ "" != "$OPA_CONNECT_TIMEOUT" ]]; then
> >   replace_key_value opaClientConfig.connectTimeOutSeconds "$OPA_CONNECT_TIMEOUT"
> > ```
> 
> Why empty check? replace_key_value already doing empty check before replacing value.

Agreed. Removed the check!


> _[`platform-service/chart/templates/config.yaml` line 30 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewODWw4hY7fLoEmZ6r:-NewODWw4hY7fLoEmZ6s:b-u6tvs2) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/chart/templates/config.yaml#L30)):_
> 
> > ```yaml
> >   { { - if .Values.global.opa.enabled } }
> >   OPA_SERVER_BASEURL: 'http://policy-mgmt.{{ .Release.Namespace }}.svc.cluster.local:3000/'
> >   OPA_SERVER_SECRET: "dOkdsVqdRPPRJG31XU0qY4MPqmBBMk0PTAGIKM6O7TGqhjyxScIdJe80mwh5Yb5zF3KxYBHw6B3Lfzlq"
> > ```
> 
> how this secret will get override?

TODO



> _[`platform-service/service/src/main/java/io/harness/platform/PlatformConfiguration.java` line 122 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewOauN9oWzZnZAMXk6:-NewOauN9oWzZnZAMXk7:b6hv1xa) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/service/src/main/java/io/harness/platform/PlatformConfiguration.java#L122)):_
> 
> > ```java
> >   @ConfigSecret private String policyManagerSecret;
> >   private ServiceHttpClientConfig opaClientConfig;
> >   @JsonProperty("enforcementClientConfiguration") private EnforcementClientConfiguration enforcementClientConfiguration;
> > ```
> 
> No @JsonProperty mapped for these configurations?

No need. Default config is needed, and replace_config fills this default configuration.

> _[`platform-service/service/src/main/resources/localhost.pem` line 21 at r3](https://reviewable.io/reviews/harness/harness-core/51753#-NewOrjJ5debhbl7pcOF:-NewOrjJ5debhbl7pcOG:bdfrdvi) ([raw file](https://github.com/harness/harness-core/blob/881f766598c545223a46d8c637f60e0b0b7b96d7/platform-service/service/src/main/resources/localhost.pem#L21)):_
> 
> > ```
> > A59Y218O3g/9MH3UVc2J7UB68N41GzYxwjk7EQn6P10EaVUqTwtjXfAxEI84UXpj
> > SBXC7VCUjCXqgsHIoGyWhBY70P04
> > -----END CERTIFICATE-----
> > ```
> 
> Why this file?

Without this file, the abstractHttpClient (OPA client) is unable to connect and fails with java cert error
Other examples are https://github.com/harness/harness-core/blob/develop/120-ng-manager/src/main/resources/localhost.pem and https://github.com/harness/harness-core/blob/develop/pipeline-service/service/src/main/resources/localhost.pem

File: platform-service/service/src/main/resources/localhost.pem (hunk 8 above)
Comment (yagyanshbhatia) [reply platform-service/service/src/main/resources/localhost.pem:1]: Without this file, the abstractHttpClient (OPA client) is unable to connect and fails with java cert error
Other examples are https://github.com/harness/harness-core/blob/develop/120-ng-manager/src/main/resources/localhost.pem and https://github.com/harness/harness-core/blob/develop/pipeline-service/service/src/main/resources/localhost.pem

File: platform-service/chart/values.yaml (hunk 7 above)
Comment (yagyanshbhatia) [reply platform-service/chart/values.yaml:30]: Done

File: platform-service/chart/templates/config.yaml (hunk 6 above)
Comment (yagyanshbhatia) [reply platform-service/chart/templates/config.yaml:31]: okay

File: platform-service/chart/templates/config.yaml (hunk 5 above)
Comment (yagyanshbhatia) [reply platform-service/chart/templates/config.yaml:28]: Done, thanks!

**Review by arya-harness — DISMISSED** [review]

**Comment by yagyanshbhatia** [comment]
trigger uts

**Comment by yagyanshbhatia** [comment]
trigger uts

**Comment by yagyanshbhatia** [comment]
trigger uts

**Comment by yagyanshbhatia** [comment]
trigger uts

**Comment by yagyanshbhatia** [comment]
trigger gitleaks, uts

**Comment by yagyanshbhatia** [comment]
trigger uts

**Comment by yagyanshbhatia** [comment]
git uts

**Comment by yagyanshbhatia** [comment]
trigger uts

**Pull request closed without merging**